### DISCUSSION TITLE
Ast Serializer for Server side manipulation

### DISCUSSION THREAD

**Discussion by brsanthu** [eclipse-langium/langium#1485] [Q&A] [answered]
We have a requirement where we need to modify user submitted SQL query to make some changes, then create new SQL statement string and execute that.

[Langium Formatting](https://langium.org/guides/formatting/) seems to be designed with Editor formatting in mind but in our case it is not applicable as it is completely server side processing.

Do you have any suggestion on how we can serialize the AST in server side?

**Comment by msujew** [comment, marked as answer]
Hey @brsanthu,

we usually just build a code generator that produces DSL code from the AST.
